server/utils/passport.ts

```diff
@@ -46,24 +47,7 @@ export class StateStore {
       );
     }
 
-    const { host, token } = parseState(state);
-
-    // Oauth callbacks are hard-coded to come to the apex domain, so we
-    // redirect to the original app domain before attempting authentication.
-    // If there is an error during auth, the user will end up on the same domain
-    // that they started from.
-    const appDomain = parseDomain(host);
-    if (appDomain.host !== parseDomain(ctx.hostname).host) {
-      const reqProtocol = ctx.protocol;
-      const requestHost = ctx.get("host");
-      const requestPath = ctx.originalUrl;
-      const requestUrl = `${reqProtocol}://${requestHost}${requestPath}`;
-      const url = new URL(requestUrl);
-
-      url.host = appDomain.host;
-
-      return callback(AuthRedirectError(``, url.toString()), false, token);
-    }
```

Comment:
This was always a pretty hacky but straightforward way to keep the user on the same subdomain, but it doesn't work at all for custom domains.